rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/backend/connection/auth/HelpGenerateUserTokenParams.java

```diff
@@ -20,14 +20,59 @@
 package org.sonarsource.sonarlint.core.rpc.protocol.backend.connection.auth;
 
 
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
 public class HelpGenerateUserTokenParams {
   private final String serverUrl;
 
+  private final Utm utm;
+
   public HelpGenerateUserTokenParams(String serverUrl) {
+    this(serverUrl, null);
+  }
+
+  public HelpGenerateUserTokenParams(String serverUrl, @Nullable Utm utm) {
     this.serverUrl = serverUrl;
+    this.utm = utm;
   }
 
   public String getServerUrl() {
     return serverUrl;
   }
+
+  @CheckForNull
+  public Utm getUtm() {
+    return utm;
+  }
+
+  public static class Utm {
+    private final String type;
+    private final String source;
```

Comment:
We have this information in many different forms in the backend, for telemetry or for the embedded server's status. But maybe it's safer to have it here too if they need to differ at some point